GradesAudit/Models/AuditRecord.cs

```diff
@@ -0,0 +1,20 @@
+using System.ComponentModel.DataAnnotations;
+
+namespace GradesAudit.Models;
+
+public class AuditRecord
+{
+    public int Id { get; set; }
+
+    [Required(ErrorMessage = "Please enter student name")]
+    [StringLength(30, ErrorMessage = "Length of student name must by lower than 30")]
+    public string StudentName { get; set; } = "";
+
+    [Required(ErrorMessage = "Please enter subject")]
+    [StringLength(15, ErrorMessage = "Subject length must by lower than 15")]
+    public string Subject { get; set; } = "";
+
+    [Required(ErrorMessage = "Please enter assessment")]
+    [Range(2, 5, ErrorMessage = "Assessment must be from 2 to 5")]
+    public uint Assessment { get; set; }
```

Comment:
```suggestion
    public uint? Assessment { get; set; }
```
Для типа значений атрибут Required не нужен, они автоматически проверяются на непустоту. Однако и выводимым сообщением в случае пустой строки управлять нельзя.
![image](https://user-images.githubusercontent.com/3691630/209588923-7c6d4a45-db69-46fb-b52a-e0fc9dbc22c1.png)
Самое простое, как это поправить - сделать тип ссылочным насильно.